imqsauth/http.go

```diff
@@ -619,6 +628,63 @@ func httpHandlerSetGroupRoles(central *ImqsCentral, w http.ResponseWriter, r *ht
 		central.Central.Log.Warnf("Group '%v' not found: %v", groupname, e)
 		authaus.HttpSendTxt(w, http.StatusNotFound, fmt.Sprintf("Group '%v' not found: %v", groupname, e))
 	}
+
+	// get all identities and their groups
+	identities, err := central.Central.GetAuthenticatorIdentities()
+	if err != nil {
+		central.Central.Log.Errorf("Error reading identities: %v\n", err)
+		return
+	}
+
+	// now find all identities which belong to the group in question,
+	// skipping the user that made the request
+	modifiedIdentities := make([]string, 0, 10)
+
+	for _, identity := range identities {
+		if r.token.Identity == identity {
+			// skip
+			continue
+		}
+		roles := getRolesList(central, identity)
+		if containsElement(groupname, roles) {
+			modifiedIdentities = append(modifiedIdentities, identity)
+		}
+	}
+	if len(modifiedIdentities) > 0 {
+		central.sendAll("auth:logout:" + strings.Join(modifiedIdentities, " "))
+	}
+}
+
+func containsElement(elem string, list []string) bool {
+	for _, s := range list {
+		if s == elem {
+			return true
+		}
+	}
+	return false
+}
+
+func getRolesList(central *ImqsCentral, identity string) []string {
```

Comment:
Remember that Go has no function overloading, so in general the function names need to be more explicit. I would call this `getRolesOfIdentity` instead.

The inner `return roles` is redundant.

Why do the `strings.Join()`, immediately followed by `strings.Split()`?

The naming is out here. The word 'roles' is confused with groups. This function should be called "getIdentityGroups". What is being returned from this function is not roles, but group names.
However, I suspect it's not really necessary to translate the group names into strings. Why not just deal directly with the group IDs (type `GroupIDU32`)? It's more efficient, and less code.